File: infrastructure/lib/oddscraper-stack.js
const { Stack } = require('aws-cdk-lib');
const lambda = require('aws-cdk-lib/aws-lambda');
const apigateway = require('aws-cdk-lib/aws-apigateway');
const secretsmanager = require('aws-cdk-lib/aws-secretsmanager')
const iam = require('aws-cdk-lib/aws-iam')

class OddscraperStack extends Stack {

  constructor(scope, id, props) {
    super(scope, id, props);

    const secret = new secretsmanager.Secret(this, 'OddscraperSecretsManager');

    const fnRole = new iam.Role(this, 'OddscraperLambdaRole', {
      assumedBy: new iam.ServicePrincipal('lambda.amazonaws.com')
    });

    secret.grantRead(fnRole);
    
    const fn = new lambda.Function(this, 'OddscraperLambda', {
      code: lambda.Code.fromAsset('../target/oddscraper-1.0.0-SNAPSHOT.jar'),
      runtime: lambda.Runtime.JAVA_21,
      handler: 'com.nathanthomp.oddscraper.OddscraperRequestHandler::handleRequest',
      role: fnRole
    });

    const endpoint = new apigateway.LambdaRestApi(this, 'OddscraperApi', {
      handler: fn,
    });

  }
}

module.exports = { OddscraperStack }

File: src/main/java/com/nathanthomp/oddscraper/OddscraperStack.java
package com.nathanthomp.oddscraper;

import software.constructs.Construct;
import software.amazon.awscdk.Stack;
import software.amazon.awscdk.StackProps;
import software.amazon.awscdk.services.apigateway.LambdaRestApi;
import software.amazon.awscdk.services.lambda.Function;

public class OddscraperStack extends Stack {
    public OddscraperStack(final Construct scope, final String id) {
        this(scope, id, null);
    }

    public OddscraperStack(final Construct scope, final String id, final StackProps props) {
        super(scope, id, props);

        Function hello = Function.Builder.create(this, "MyFunction")
                .runtime(software.amazon.awscdk.services.lambda.Runtime.NODEJS_LATEST)
                .code(software.amazon.awscdk.services.lambda.Code.fromAsset("lib/lambda-handler"))
                .handler("index.handler")
                .build();

        LambdaRestApi api = LambdaRestApi.Builder.create(this, "ApiGwEndpoint")
                .restApiName("HelloApi")
                .handler(hello)
                .build();

    }
}